### PR TITLE
Look for settings files at '.nfront/command.ext'

### PR DESCRIPTION
**Pull request by ForbesLindesay**
Searches for command.json, commant.yml and command.yaml and moves up the directory tree all the way to the top, merging in settings so the local ones override the global ones.

### CODE REVIEW

File: nodefront.js [hunk 1]
```diff
@@ -17,7 +18,7 @@ program
   .option('-r, --recursive', 'Recurse through directories.')
   .option('-w, --watch', 'Recompile files upon modification. Intelligently' +
     ' recompile dependencies.')
-  .action(require('./commands/compile'));
+  .action(config.withSettings('./.nfront/compile', require('./commands/compile')));
```
Comment (karthikv) [inline nodefront.js]: Could we change the config directory to `.nf`? That's the shorthand I've been using so far for nodefront.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
     "request": "2.9.x",
     "socket.io": "0.9.x",
     "stylus": "0.27.x",
-    "uglify-js": "1.3.x"
+    "uglify-js": "1.3.x",
+    "commander-config": "0.x.x"
```
Comment (karthikv) [inline package.json]: Dependencies are in alphabetical order for ease of scanning. Could you move this one to its appropriate location?

**Comment by karthikv** [comment]
Looks good! I commented on some minor issues; let me know when they are fixed and I'll merge in!

**Comment by ForbesLindesay** [comment]
Fixed

**Comment by karthikv** [comment]
Great work! Your changes have been integrated.

**Pull request closed without merging**